Update the readme file (#15976)
DataDog/integrations-core: ray/README.md

Author: ✅

### Installation

Starting from Agent release 7.49.0, the Ray check is included in the [Datadog Agent][2] package. No additional installation is needed on your server.

> <div class="alert alert-warning">This check uses <a href="https://docs.datadoghq.com/integrations/openmetrics/">OpenMetrics</a> to collect metrics from the OpenMetrics endpoint Ray can expose, which requires Python 3.</div>

Reviewer: ```suggestion
**WARNING**: This check uses [OpenMetrics](https://docs.datadoghq.com/integrations/openmetrics/) to collect metrics from the OpenMetrics endpoint Ray can expose, which requires Python 3.
```
This is a fairly new discovery on our side, but these alert boxes are stripped out of the app tiles, so it's best just to use standard markdown to make sure they're visible there.

Author: ✅

      - openmetrics_endpoint: http://<RAY_ADDRESS>:8080
    ```

2. [Restart the Agent][5] after modifying the configuration.

<!-- xxz tab xxx -->
<!-- xxx tab "Docker" xxx -->

#### Docker

##### Metric collection

This example demonstrates the configuration as a Docker label inside `docker-compose.yml`. See the [sample ray.d/conf.yaml][4] for all available configuration options.

```yaml
labels:
  com.datadoghq.ad.checks: '{"ray":{"instances":[{"openmetrics_endpoint":"http://%%host%%:8080"}]}}'
```

<!-- xxz tab xxx -->
<!-- xxx tab "Kubernetes" xxx -->

#### Kubernetes

##### Metric collection

This example demonstrates the configuration in the previous sections as Kubernetes annotations on your Ray pods. See the [sample ray.d/conf.yaml][4] for all available configuration options.

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: '<POD_NAME>'
  annotations:
    ad.datadoghq.com/ray.checks: |-
      {
        "ray": {
          "instances": [
            {
              "openmetrics_endpoint": "http://%%host%%:8080",
            }
          ]
        }
      }
    # (...)
spec:
  containers:
    - name: 'ray'
# (...)
```

<!-- xxz tab xxx -->
<!-- xxz tabs xxx -->

> Ray allows the custom service code to emit [your own metrics that will be available on the OpenMetrics endpoint][10]. You can configure this integration to collect these metrics using the `extra_metrics` option. These metrics will have the `ray.` prefix, just like any other metrics coming from this endpoint.

Reviewer: ```suggestion
Ray metrics are available on the OpenMetrics endpoint. Additionally, Ray allows you to [export custom application-level metrics][10]. You can configure the Ray integration to collect these metrics using the `extra_metrics` option. All Ray metrics, including your custom metrics, use the `ray.` prefix.
```

> <div class="alert alert-info">These custom Ray metrics are considered standard metrics in Datadog.</div>

Reviewer: ```suggestion
**Note:** Custom Ray metrics are considered standard metrics in Datadog.
```

This example demonstrates a configuration leveraging the `extra_metrics` option:

```yaml
init_config:
  ...
instances:
  - openmetrics_endpoint: http://<RAY_ADDRESS>:8080
    # Also collect your own Ray metrics
    extra_metrics:
      - my_custom_ray_metric
```

> More info on how to configure this option can be found in the [sample ray.d/conf.yaml][11].

Reviewer: ```suggestion
More info on how to configure this option can be found in the [sample `ray.d/conf.yaml` configuration file][11].
```

[7]: https://github.com/DataDog/integrations-core/blob/master/ray/metadata.csv
[8]: https://github.com/DataDog/integrations-core/blob/master/ray/assets/service_checks.json
[9]: https://docs.datadoghq.com/help/
[10]: https://docs.ray.io/en/latest/ray-observability/user-guides/add-app-metrics.html
[11]: https://github.com/DataDog/integrations-core/blob/master/ray/datadog_checks/ray/data/conf.yaml.example#L59-L105 
[12]: https://docs.ray.io/en/latest/ray-observability/user-guides/configure-logging.html

> [13]: https://docs.datadoghq.com/agent/kubernetes/log/?tab=containerinstallation#setup

Reviewer: ```suggestion
[13]: https://docs.datadoghq.com/agent/kubernetes/log/#setup
```
